### PR TITLE
MODSOURMAN-794 - Removed API /metadata-provider/logs/{jobExecutionId} and related files

### PR DESCRIPTION
**Pull request by SatishkumarPurswani**
## Purpose
API - /metadata-provider/logs/{jobExecutionId} was not required and was not being used.

## Approach
Removed the APIs related code, test cases, and files

## Learning
Have created this PR irrespective of MODSOURMAN-785 as the change are not much related to the same Jira but can only be referenced.

### CODE REVIEW

File: mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/DataImportJournalConsumerVerticleTest.java [hunk 1]
```diff
@@ -101,167 +95,6 @@ public void setUp() {
...
-        .withContentType(ACTION_PROFILE)
-        .withContent(JsonObject.mapFrom(new ActionProfile().withFolioRecord(ActionProfile.FolioRecord.HOLDINGS))))
-      .withEventsChain(List.of(DI_SRS_MARC_BIB_RECORD_CREATED.value(), DI_INVENTORY_HOLDING_CREATED.value()));
-
-    // when
-    KafkaConsumerRecord<String, String> kafkaConsumerRecord = buildKafkaConsumerRecord(eventPayload);
-    dataImportJournalKafkaHandler.handle(kafkaConsumerRecord);
-
-    // then
-    Future<JobExecutionLogDto> future = journalRecordDao.getJobExecutionLogDto(jobExecution.getId(), TENANT_ID);
```
Comment (SerhiiNosko) [inline mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/DataImportJournalConsumerVerticleTest.java:257]: I think we don't need to delete these test methods, just need to replace getJobExecutionLogDto() to getByJobExecutionId()

**Comment by okolawole-ebsco** [comment]
Why is this endpoint being removed? what endpoint do we use to retrieve a log after this is removed? or we dont retrieve logs anymore?

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

File: mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/DataImportJournalConsumerVerticleTest.java (hunk 1 above)
Comment (SatishkumarPurswani) [reply mod-source-record-manager-server/src/test/java/org/folio/verticle/consumers/DataImportJournalConsumerVerticleTest.java:257]: Hi @SerhiiNosko - Updated, Thank you 🙂

**Comment by SatishkumarPurswani** [comment]
> Why is this endpoint being removed? what endpoint do we use to retrieve a log after this is removed? or we dont retrieve logs anymore?

Hi @okolawole-ebsco - 
API - /metadata-provider/logs/{jobExecutionId} is not required and was not being used.
We have another API to retrieve logs - /change-manager/jobExecutions/{id}

**Comment by okolawole-ebsco** [comment]
> > Why is this endpoint being removed? what endpoint do we use to retrieve a log after this is removed? or we dont retrieve logs anymore?
> 
> Hi @okolawole-ebsco - API - /metadata-provider/logs/{jobExecutionId} is not required and was not being used. We have another API to retrieve logs - /change-manager/jobExecutions/{id}

Thanks! was this endpoint added in the release? if it was added a lot earlier, then this will constitute a contract change. I think product management would need to be notified to take next steps.

**Review by okolawole-ebsco — APPROVED** [review]

**Pull request merged**